### PR TITLE
Update current SwiftPM version to 5.5

### PR DESCRIPTION
**Pull request by abertelrud**
This updates the current SwiftPM to 5.5 (see https://forums.swift.org/t/swift-5-5-release-process/45644).

### Motivation:

When a new Swift release is announced, SwiftPM is updated to reflect that version.

### Modifications:

- change default version to 5.5.0
- add `v5_5` tools version
- modify `999.0` availability and `vNext` to `5.5` and `v5_5` respectively
- adjust unit test to reflect that executable modules are now linkable by unit tests

Note that SwiftPM is still marked as a development version so that 999.0 is still accepted in mainline.  This will be changed after branching 5.5.

### CODE REVIEW

**Comment by abertelrud** [comment]
@swift-ci please smoke test

File: Tests/WorkspaceTests/InitTests.swift [hunk 1]
```diff
@@ -90,7 +90,7 @@ class InitTests: XCTestCase {
             let triple = Resources.default.toolchain.triple
             let binPath = path.appending(components: ".build", triple.tripleString, "debug")
             XCTAssertFileExists(binPath.appending(component: "Foo"))
-            XCTAssertFileExists(binPath.appending(components: "Foo.build", "Foo.swiftmodule"))
+            XCTAssertFileExists(binPath.appending(components: "Foo.swiftmodule"))
```
Comment (abertelrud) [inline Tests/WorkspaceTests/InitTests.swift:93]: The feature to allowing linking against executable modules changes the output path based on the tools version, and new packages now get 5.5 by default.  So the unit test changes.

**Review by neonichu — APPROVED** [review]

**Comment by abertelrud** [comment]
The error seems to be because in the self-hosted test the Swift compiler doesn't yet have the flag that we rely on, i.e. it's building using an old compiler.

`12:41:57     [1/2] Compiling Foo main.swift
12:41:57     <unknown>:0: error: unknown argument: '-entry-point-function-name'
12:41:57     , output: "[1/2] Compiling Foo main.swift\n<unknown>:0: error: unknown argument: \'-entry-point-function-name\'\n", stderr: "")`

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Comment by abertelrud** [comment]
Misread the error, this is actually the `IntegrationTests` package.  This test will need to be skipped if the compiler doesn't support the `-entry-point-function-name` option.

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Comment by abertelrud** [comment]
The self-hosted Linux failure looks like infrastructure: `java.io.EOFException`
Rerunning

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Review by tomerd — APPROVED** [review]

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Pull request merged**